include/rcutils/find.h

```diff
@@ -23,14 +23,66 @@ extern "C"
 #include "rcutils/types.h"
 #include "rcutils/visibility_control.h"
 
+/// Return the first index of a character in a string.
+/**
+ * Search in a string for the first occurence of a delimiter.
+ *
+ * \param[in] str null terminated c string to search
+ * \param[in] delimiter the character to search for
+ * \returns the index of the first occurence of the delimiter if found, or
+ * \returns `string_length` if the delimiter is not found, or
+ * \returns `0` for invalid arguments
```

Comment:
@clalancette raised the question in https://github.com/ros2/rcutils/pull/57#discussion_r145206340; it will be addressed as followup https://github.com/ros2/rcutils/issues/58